lib/IR/ChildMemSizeBasedScheduler.cpp

```diff
@@ -81,103 +81,130 @@ void ChildMemSizeBasedScheduler::computeNodeComputationMaxMemorySize() {
 /// Order children by (maxSize - resultSize). It gives more
 /// priority to the nodes that free more memory after
 /// their computation.
-void ChildMemSizeBasedScheduler::orderChildNodesAndSchedule(Node *N) {
-  // Each child should be scheduled just once.
-  if (isScheduled(N))
-    return;
+void ChildMemSizeBasedScheduler::orderChildNodesAndSchedule(Node *node) {
   // Do not explicitly schedule storage nodes.
-  if (isa<Storage>(N))
+  if (isa<Storage>(node)) {
```

Comment:
Is it really needed or will be take care of by the while-loop below anyways?

Reply:
I had some algorithm performance concerns (since `orderChildNodesAndSchedule` is called in a loop for each node), but my experiments proved there is no performance degradation if I remove these checks.
Also, it appears reserving space for worklist and set does not affect performance either. So, to avoid allocating full-graph worklist and set for nodes which might exit on the first algorithm iterations, I removed space reservation as well.